### PR TITLE
mixin: New receiver alert to flag unusually low ingestion rate

### PR DESCRIPTION
**Pull request by spaparaju**
Signed-off-by: spaparaju <paparaju@gmail.com>

* [x] I added CHANGELOG entry for this change.

## Changes

- Added new alert ThanosReceiveTrafficBelowThreshold to flag unusually low ingestion rate (ingestion average of the last hour is 50% of the observed average for the last 12 hours)

## Verification

- make examples
- make example-rules-lint

### CODE REVIEW

File: examples/alerts/alerts.md [hunk 1]
```diff
@@ -562,6 +562,23 @@ rules:
...
+  annotations:
+    description: At Thanos Receive {{$labels.job}} in {{$labels.namespace}} , the
+      average 1-hr avg. metrics ingestion rate  is {{$value | humanize}}% of 12-hr
+      avg. ingestion rate.
+    runbook_url: https://github.com/thanos-io/thanos/tree/main/mixin/runbook.md#alert-name-thanosreceivetrafficbelowthreshold
+    summary: Thanos Receive is experiencing low avg. 1-hr ingestion rate relative
+      to avg. 12-hr ingestion rate.
+  expr: |
+    (
+      avg by (job) (rate(http_requests_total{code=~"2..", job=~".*thanos-receive.*", handler="receive"}[1h]))
```
Comment (bwplotka) [inline examples/alerts/alerts.md:575]: For average within one hour we need to use sum sub query so something like this:
`avg_over_time(rate(http_requests_total{code=~"2..", job=~".*thanos-receive.*", handler="receive"}[5m])[1h:5m])` 

The `avg` makes average over series not over time.

Comment (spaparaju) [reply examples/alerts/alerts.md]: 👍  Thanks @bwplotka !

File: mixin/alerts/receive.libsonnet [hunk 2]
```diff
@@ -143,6 +144,24 @@
...
+            alert: 'ThanosReceiveTrafficBelowThreshold',
+            annotations: {
+              description: 'At Thanos Receive {{$labels.job}} in {{$labels.namespace}} , the average 1-hr avg. metrics ingestion rate  is {{$value | humanize}}% of 12-hr avg. ingestion rate.',
+              summary: 'Thanos Receive is experiencing low avg. 1-hr ingestion rate relative to avg. 12-hr ingestion rate.',
+            },
+            expr: |||
+              (
+                avg_over_time(rate(http_requests_total{code=~"2..", job=~".*thanos-receive.*", handler="receive"}[5m])[1h:5m])
+              /
+                avg_over_time(rate(http_requests_total{code=~"2..", job=~".*thanos-receive.*", handler="receive"}[5m])[12h:5m])
```
Comment (kakkoyun) [inline mixin/alerts/receive.libsonnet:157]:
```suggestion
                avg_over_time(rate(http_requests_total{%(selector)s, code=~"2..", handler="receive"}[5m])[12h:5m])
```

**Review by kakkoyun — CHANGES_REQUESTED** [review]

File: mixin/alerts/receive.libsonnet (hunk 2 above)
Comment (spaparaju) [reply mixin/alerts/receive.libsonnet]: :+1:

**Review by kakkoyun — APPROVED** [review]

**Review by squat — APPROVED** [review]
<3

**Pull request merged**